### PR TITLE
Rory Nolan ijtiff blog.

### PR DESCRIPTION
**Pull request by rorynolan**
Hi @stefaniebutland,
Here is my blog post. Thanks again for asking me to do this.
I'm away Monday-Thursday next week but can definitely respond to anything on Friday if necessary.
Regards,
Rory

### CODE REVIEW

**Comment by stefaniebutland** [comment]
@rorynolan So sorry for my delay. I will review this by Wed Mar 28 and suggest a date to publish it.

**Comment by stefaniebutland** [comment]
@rorynolan You could consider adding a "pullquote" to draw attention to a certain line, for example you might add this

> Something that will make life easier in the long-run can be the most difficult thing to do today

right before Conclusions or somewhere else where the sentiment fits

Add a pullquote by putting `> ` before the text. You can see how these appear in a post by looking at https://ropensci.org/tags/rprofile/

File: content/blog/2018-03-13-ijtiff.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+slug: "ijtiff"
+title: Forcing Yourself to Make Your Life Easier
+package_version: 1.0.0
+authors:
+  - name: Rory Nolan
+    url: https://github.com/rorynolan
+date: 2018-03-13
```
[the remark this answers is not in the document]
Comment (rorynolan) [reply content/blog/2018-03-13-ijtiff.md]: Done.

File: content/blog/2018-03-13-ijtiff.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+- software
+- review
+- onboarding
+- package
+- tiff
+- data-import
+- ImageJ
+---
+
+## The general struggle
```
[the remark this answers is not in the document]
Comment (rorynolan) [reply content/blog/2018-03-13-ijtiff.md]: Done.

File: content/blog/2018-03-13-ijtiff.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+Something that will make life easier in the long-run can be the most difficult thing to do _today_. For coders, prioritising the long term may involve an overhaul of current practice and the learning of a new skill. This can be painful for a number of reasons:
```
[the remark this answers is not in the document]
Comment (rorynolan) [reply content/blog/2018-03-13-ijtiff.md]: Thanks a lot! Very nice to hear :-)

File: content/blog/2018-03-13-ijtiff.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
.../cran.r-project.org/package=magick) packages can both read TIFF files, and on [Bioconductor](https://www.bioconductor.org/), there’s [`EBImage`](https://bioconductor.org/packages/EBImage/). However, all of these packages sometimes struggle with TIFF files written from _ImageJ_ in that they wrongly perceive some images to have only 1 channel when in fact they have many (channels encode colour information: colour images have 3 channels (red, green and blue) whereas black and white images have one channel (grey)). Once the images are read into R, I am able to rejig them (with a combination of `aperm()` and `abind()`) into the format I want. However, the mistakes that the packages make vary, and thus the images require different rejigs. Every time I wanted to read an image into R, the process was _read, check, rejig, check_. This is a lot longer than just _read_. Nonetheless, this lengthy reading process still wasn’t that long in absolute terms: I could read in an image and have it in the format I wanted in about a minute. I processed thousands of images over the following two years (with various image analysis techniques). For each one, I went through _read, check, rejig, check_ . . .
```
[the remark this answers is not in the document]
Comment (rorynolan) [reply content/blog/2018-03-13-ijtiff.md]: Done.

[the remark this answers is not in the document]
Comment (rorynolan) [reply content/blog/2018-03-13-ijtiff.md]: Done.

File: content/blog/2018-03-13-ijtiff.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+In 2016, I attended Bioconductor’s [CSAMA](http://www-huber.embl.de/csama2018/) event in Italy. [Jenny Bryan](https://www.stat.ubc.ca/~jenny/) was there advocating the [tidyverse](https://www.tidyverse.org/), [happygitwithr](http://happygitwithr.com/) and various other good ideas. At that time, I wasn’t much into the R scene, so I’d never heard of Jenny nor of many of the things she mentioned, but I was very taken by her teaching (she grabbed my attention with the tip that `Alt`+`-` in RStudio gets you the assignment operator `<-`). The advice that resonated most was that _NOW is the time to do that thing that you know you ought to do eventually._ We all know that this is good advice, but that doesn’t mean we don’t need to be told it frequently.
```
[the remark this answers is not in the document]
Comment (rorynolan) [reply content/blog/2018-03-13-ijtiff.md]: I know, I went years not knowing this!

File: content/blog/2018-03-13-ijtiff.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
... them from some random place atop every `.R` file). So I learned git, made a GitHub account and wrote my first R package. I decided to start easy, gathering together all of the functions that I use to rename and organise my files. This became (the terribly named) [`filesstrings`](https://cran.r-project.org/package=filesstrings) (now on CRAN). The most useful thing it can do is to extract numbers from strings with functions like `first_number()`. `filesstrings` used `Rcpp`, so I got comfortable writing R packages that use C++. Having completed `filesstrings`, I looked more seriously into writing a package that solved my _ImageJ_ TIFF issue. It was clear that I would require the `libtiff` C library, and thus would need to use R’s C interface, so I looked for documentation of it. The section in Hadley Wickham’s [_Advanced R_](http://adv-r.had.co.nz/C-interface.html) and his GitHub repo [r-internals](https://github.com/hadley/r-internals) are both good resources, but I still feel that this facet of R is under-documented. . As a result, I couldn’t really get to grips with R’s C interface (at least not within a week as I’d hoped), so I gave up and returned to _read, check, rejig, check_.
```
[the remark this answers is not in the document]
Comment (rorynolan) [reply content/blog/2018-03-13-ijtiff.md]: Fixed.

File: content/blog/2018-03-13-ijtiff.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+## Eventually
+
+In the end, I did write `ijtiff`—the package I had been longing for—using `libtiff` and R’s C interface. What happened that pushed me over the edge to learn everything I needed to know about C and R’s interface to it and to slave away for six weeks with `clang`  errors and RStudio crashes and `valgrind`  in Linux virtual machines? Nothing in particular. I was doing yet another _read, check, rejig, check_  and I had had enough. I decided to take another crack at `ijtiff`. This time, with the help of what I’d learned during all of my previous failed attempts, I made it over the line.
```
[the remark this answers is not in the document]
Comment (rorynolan) [reply content/blog/2018-03-13-ijtiff.md]: Thanks for the pointer about the slave -> toil. Much more PC. It's the type of thing one says without even realising.

File: content/blog/2018-03-13-ijtiff.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+## Immediate future
+
+As mentioned previously, there is a lack of accessible information on R's C interface. The demand on a few experts like [Hadley Wickham](http://hadley.nz/), [Kevin Ushey](https://kevinushey.github.io/), [Jim Hester](https://github.com/jimhester) etc. to provide resources for the rest of us is huge. Although these people are very generous and well-appreciated, they only have a finite amount of time. As such, I want to contribute to the documentation on R's C interface. 
+
+## Conclusions
+
+1.  Find and heed all of [Jenny Bryan](https://www.stat.ubc.ca/~jenny/)’s advice.
```
[the remark this answers is not in the document]
Comment (rorynolan) [reply content/blog/2018-03-13-ijtiff.md]: Done.

File: content/blog/2018-03-13-ijtiff.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+3.  If it's a lack of skills that prevents you from doing what you wish you could do, take the time to learn those skills. They’ll make your life easier again and again in future.
+4.  If you figure out how to do something that you wish was made easier in documentation, please—for the likes of me—contribute that new knowledge to existing documentation.
+
+### Thanks
+
+* Thanks to [ROpenSci](https://ropensci.org/) for their very helpful review of the `ijtiff` package. This type of review where the reviewers actively help you as well as objectively evaluating your work is a revelation. Getting the advice and help of people like [Jon Clayden](https://github.com/jonclayden) and [Jeroen Ooms](https://github.com/jeroen) was invaluable. Thanks also to editor [Scott Chamberlain](https://github.com/sckott) and to community manager [Stefanie Butland](https://github.com/stefaniebutland) who is helping me with this blog.
```
[the remark this answers is not in the document]
Comment (rorynolan) [reply content/blog/2018-03-13-ijtiff.md]: Capitalization fixed. I've left the thanks to you in there. You've been lovely to work with :-)

**Comment by rorynolan** [comment]
Thanks a lot for all the comments @stefaniebutland. I agree on all counts and have made the changes.
You made some great points which I really appreciate and I also love your encouragement :-)

**Comment by stefaniebutland** [comment]
Thanks for the updates @rorynolan, and for the thanks. Could you please change my link to https://twitter.com/StefanieButland? Reflects better on me than my GitHub ;-)

**Comment by rorynolan** [comment]
Done :-)

**Comment by stefaniebutland** [comment]
@rorynolan A blog spot opened up. Would you please update the date to 2018-04-12? 

Since post does not have images, looks like that's the only update needed. After you update, I will add the topicid and publish and then tweet from rOpenSci

**Comment by rorynolan** [comment]
Done. Thanks so much! Excited to get it out there :-)

**Comment by rorynolan** [comment]
Hi @stefaniebutland, thanks for all your kind work promoting my blog. I'm delighted with how it's been received and I really appreciate your efforts in getting it noticed :-)

**Pull request merged**